kaldb/src/main/java/com/slack/kaldb/logstore/search/LogIndexSearcherImpl.java

```diff
@@ -51,9 +55,17 @@
 public class LogIndexSearcherImpl implements LogIndexSearcher<LogMessage> {
   private static final Logger LOG = LoggerFactory.getLogger(LogIndexSearcherImpl.class);
 
+  // number of threads to use transforming the results
+  private static final int PARALLEL_THREADS = 12;
+
   private final SearcherManager searcherManager;
   private final StandardAnalyzer analyzer;
 
+  private final ExecutorService executorService =
+      Executors.newFixedThreadPool(
```

Comment:
Why not use a cached thread pool?

Reply:
We could, but there isn't expected to be a meaningful difference. We use this threadpool for every query, so in order for a cached threadpool to reduce the threads it would need to have no queries that had more than 12 results for 60 seconds.